proposals/0216-widget-support.md

```diff
@@ -180,69 +270,61 @@ Details about the currently selected screen template are part of the display cap
-
-#### Screen related `OnHMIStatus`
-
-In the proposal widget screens should not follow HMI levels. RPCs sent to a widget screen would follow app policies of HMI_LIMITED. This alternative proposes OnHMIStatus for any screen instance.
-
-```xml
-<function name="OnHMIStatus" functionID="OnHMIStatusID" messagetype="notification" since="1.0">
-  :
-  <param name="screenID" type="Integer" mandatory="false" since="5.1" />
-</function>    
-```
+## Potential downsides
 
-The additional parameter can be used by the system to tell the app the HMI level of any screen instance. The parameter should be omitted for the predefined default screen for better backward compatibility. For other screens the screen ID specified by the app should be set in this parameter.
+No downsides identified.
 
-By default, widgets created are in HMI_BACKGROUND. This allows the app to fill the widget with content after it's created. If they get visible on the display, the HMI should notify the app that the widget is now in HMI_FULL. This can be the case if the driver selects the widget area (e.g. the home screen shows app widgets). As widgets are not launchable they should automatically transit to HMI_FULL once they are visible.
+## Impact on existing code
 
-To limit the changes in the RPC and support backward compatibility, audioStreamingState should be provided to all screens of one app. The HMI and SDL core must make sure to not send different audio states per app.
+There are additional RPCs `CreateScreen` and `DeleteScreen` that need to be added. In addition to this `Show`, `SetDisplayLayout` and `OnHMIStatus` need to be extended with a `screenID` parameter.
 
-System context can be screen related. A widget can be fully visible (system context MAIN) while the app's default screen can present the menu (system context MENU).
+After investigating impact to SDL Core, the impact is expected to be minor. Global approach remains the same for now. The following items have been identified as affected by this proposal:
+1. Changes to RPC service as new RPCs and parameters are introduced
+2. Changes to State Controller to allow HMI level transitions for widgets and screens
+3. Changes to Request Controller as RPCs from one app can be addressed to different screens
 
-#### Screen duplication by SDL libraries
+## Alternatives considered
 
-To reduce complexity on the head unit, screen duplication can be supported by the SDL libraries. This could allow more flexibility to enable or display duplication for existing screens. However this increases the number of RPCs to be sent by the app and the user might see a delay in screen update (duplication not synchronized).
+To reduce complexity on the head unit, screen duplication can be supported by the SDL libraries. This could allow more flexibility to  display duplication for existing screens. However this increases the number of RPCs to be sent by the app and the user might see a delay in screen update (duplication not synchronized).
```

Comment:
Update to:
To reduce complexity on the head unit, screen duplication can be supported by the SDL libraries. This could allow more flexibility to  display duplication for existing screens. However this increases the number of RPCs to be sent by the app and the user might see a delay in screen **updates** (duplication not synchronized).